src/Http/Cookie/Cookie.php

```diff
@@ -0,0 +1,506 @@
+    {
+        $headerValue[] = sprintf('%s=%s', $this->name, urlencode($this->value));
+
+        if ($this->expiresAt !== 0) {
+            $headerValue[] = $this->_buildExpirationValue();
+        }
+        if (!empty($this->path)) {
+            $headerValue[] = sprintf('path=%s', $this->path);
+        }
+        if (!empty($this->domain)) {
+            $headerValue[] = sprintf('domain=%s', $this->domain);
+        }
+        if ($this->secure) {
+            $headerValue[] = 'secure';
+        }
+        if ($this->httpOnly) {
+            $headerValue[] = 'httponly';
+        }
+
+        return implode('; ', $headerValue);
+    }
+
+    /**
+     * Sets the cookie name
+     *
+     * @param string $name Name of the cookie
+     * @return $this
+     */
+    public function setName($name)
+    {
+        $this->validateName($name);
+        $this->name = $name;
+
+        return $this;
+    }
+
+    /**
+     * Gets the cookie name
+     *
+     * @return string
+     */
+    public function getName()
+    {
+        return $this->name;
+    }
+
+    /**
+     * Validates the cookie name
+     *
+     * @param string $name Name of the cookie
+     * @return void
+     * @throws \InvalidArgumentException
+     */
+    protected function validateName($name)
+    {
+        if (preg_match("/[=,; \t\r\n\013\014]/", $name)) {
+            throw new InvalidArgumentException(
+                sprintf('The cookie name `%s` contains invalid characters.', $name)
+            );
+        }
+
+        if (empty($name)) {
+            throw new InvalidArgumentException('The cookie name cannot be empty.');
+        }
+    }
+
+    /**
+     * Gets the raw cookie value
+     *
+     * @return string
+     */
+    public function getValue()
+    {
+        return $this->value;
+    }
+
+    /**
+     * Sets the raw cookie data
+     *
+     * @param string|array $value Value of the cookie to set
+     * @return $this
+     */
+    public function setValue($value)
+    {
+        if (is_array($value)) {
+            $this->isExpanded = true;
+        }
+
+        $this->value = $value;
+
+        return $this;
+    }
+
+    /**
+     * Sets the path
+     *
+     * @param string|null $path Sets the path
+     * @return $this
+     */
+    public function setPath($path)
+    {
+        $this->path = $path;
+
+        return $this;
+    }
+
+    /**
+     * Sets the domain
+     *
+     * @param string $domain Domain to set
+     * @return $this
+     */
+    public function setDomain($domain)
+    {
+        $this->domain = $domain;
+
+        return $this;
+    }
+
+    /**
+     * Check if the cookie is secure
+     *
+     * @return bool
+     */
+    public function isSecure()
+    {
+        return $this->secure;
+    }
+
+    /**
+     * Set HTTP Only
+     *
+     * @param bool $httpOnly HTTP Only
+     * @return $this
+     */
+    public function setHttpOnly($httpOnly)
+    {
+        if (!is_bool($httpOnly)) {
```

Comment:
Could there be a check like  in other setters that require strings? Is it necessary? Or maybe just cast `(bool)$httpOnly` here?